### DISCUSSION TITLE
Will VisTrace be getting any comprehensive documentation?

### DISCUSSION THREAD

**Discussion by yogwoggf** [Derpius/VisTrace#95] [Ideas]
I'd like to know if VisTrace will be getting any comprehensive documentation. Currently, you kind of have to go by the source code which is majorly unintuitive.

@Derpius 
Creating a automatic docgen would be pretty easy if we were to comment above each of the `LUA_FUNCTION` declarations in the main file and then scrape that and put it into some machine-readable format (for EmmyLua bindings, etc.) and a human-readable format for the normal VisTrace user.

I get that there's not much time for VisTrace, but seeing as it actually has a decent userbase, I think it'd be worth looking into cleaning it up a bit. For example:

1. Examples, only information about this was a lousy "moving to dedicated branch," and nothing else.
2. Of course, documentation
3. Could use the GitHub wiki, but I don't know if you could automatically generate it.

Anyways, I'd appreciate if the documentation can be in machine-readable format readily accessible for anyone who wants to scrape it and do stuff with it (me)

**Comment by Derpius** [comment]
Did you use GPT for this? This is far too coherent compared to your usual schizophrenic rambling…

Anyway given the very small amount of available dev time I have, and that both Vis
Tracing in one weekend and the Starfall documentation exists, additional docs and proper examples are very much low priority.

if you want to work on a docgen system then go ahead, I don’t like the idea of multiple sources of truth between the GLua and StarfallEx docs but that’s kinda unavoidable. No need to include it in the C++ source, can just do what Nanos World does and have it as JSON.
